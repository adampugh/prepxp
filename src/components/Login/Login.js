import React, { Component } from "react";
import { connect } from "react-redux";

import LoginBox from "./loginBox";
import SignupBox from "./signupBox";
import Footer from "../UI/footer";
import * as actions from "../../store/actions/auth";

export class Login extends Component {
    state = {
        login: true
    }

    componentDidMount () {
        window.scrollTo(0, 0);
    }

    switchToLogin = () => {
        this.setState({
            login: !this.state.login
        });
    }

    validateEmail = (email) => {
        var re = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
        return re.test(email);
    }

    render() {
        return (
            <div>
            <div className="login">
                <h1 className="logo">PrepXP</h1>
                {this.state.login 
                    ? <LoginBox 
                        switchToLogin={this.switchToLogin}
                        startLogin={this.props.startLogin} 
                        validateEmail={this.validateEmail} /> 
                    : <SignupBox 
                        switchToLogin={this.switchToLogin} 
                        startSignUp={this.props.startSignUp} 
                        validateEmail={this.validateEmail} />}
            </div>
            <h1 onClick={this.props.startLogout}>Logout</h1>
            <Footer />
        </div>
            
        )
    }
}

const mapDispatchToProps = dispatch => {
    return {
        startSignUp: (email, password, name) => dispatch(actions.startSignUp(email, password, name)),
        startLogin: (email, password) => dispatch(actions.startLogin(email, password)),
        startLogout: () => dispatch(actions.startLogout())
    }
}

export default connect(null, mapDispatchToProps)(Login);