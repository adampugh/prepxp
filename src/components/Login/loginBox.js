import React, { Component } from "react";

class loginBox extends Component {
    state = {
        error: false
    }

    handleLogin = () => {
        const email = this.refs.emailLogin.value;
        const password = this.refs.passwordLogin.value;
        this.props.startLogin(email, password)
            .catch((e) => {
                this.setState({
                    error: true
                })
            })
    }

    render() {
        return (
            <div>
                <div className="login__box">
                    <h2>Log in to your account</h2>
                    <p>{this.state.error ? "Sorry there was an error" : null}</p>
                    <div className="login__box__input">
                        <input type="email" placeholder="email" ref="emailLogin"/>
                        <i className="fas fa-envelope"></i>
                    </div>
                    <div className="login__box__input">
                        <input type="password" placeholder="password" ref="passwordLogin" />
                        <i className="fas fa-lock"></i>
                    </div>
                    <button 
                        onClick={this.handleLogin}
                        className="btn">Log In</button>
                </div>
                <div className="login__bottomBox">
                    <p>New to PrepXP? <span onClick={this.props.switchToLogin}>Sign up</span></p>
                </div>
            </div>
        );
    }
}

export default loginBox;