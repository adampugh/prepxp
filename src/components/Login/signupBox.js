import React, { Component } from "react";

class signupBox extends Component {
    state = {
        error: false,
        errorMessage: ""
    }

    handleStartSignUp = () => {
        const name = this.refs.nameInput.value;
        const email = this.refs.emailInput.value;
        const password = this.refs.passwordInput.value;

        // validate user input
        if (name.length < 1) {
            this.setState({
                error: true,
                errorMessage: "Please enter a name"
            });
        } else if (!this.props.validateEmail(email)) {
            this.setState({
                error: true,
                errorMessage: "Please enter a valid email address"
            })
        } else if (password.length < 6) {
            this.setState({
                error: true,
                errorMessage: "Please enter a password with more than 6 characters"
            })
        } else {
            this.props.startSignUp(email, password, name).catch((e) => {
                this.setState({
                    error: true,
                    errorMessage: e.message
                })
            })
        }        
    }

    render() {
        return (
            <div>
                <div className="login__box">
                    <h2>Sign up for an account</h2>
                    <p className="error">{this.state.error ? this.state.errorMessage : null}</p>
                    <div className="login__box__input">
                        <input type="text" placeholder="name" ref="nameInput"/>
                        <i className="fas fa-user"></i>
                    </div>
                    <div className="login__box__input">
                        <input type="email" placeholder="email" ref="emailInput"/>
                        <i className="fas fa-envelope"></i>
                    </div>
                    <div className="login__box__input">
                        <input type="password" placeholder="password" ref="passwordInput"/>
                        <i className="fas fa-lock"></i>
                    </div>
                    <button
                        onClick={this.handleStartSignUp} 
                        className="btn">
                            Sign Up
                    </button>
                </div>
                <div className="login__bottomBox">
                    <p>Already have an account? <span onClick={this.props.switchToLogin}>Log in</span></p>
                </div>
            </div>
        );
    }
} 

export default signupBox;