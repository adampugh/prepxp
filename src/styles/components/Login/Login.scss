.login {
    text-align: center;
    background: $purple-gradient, url("../assets/images/bg4.jpg") no-repeat center;
    background-size: cover;
    // display: grid;
    height: 100vh; 
    padding: $padding-m 10px;
    box-sizing: border-box;
    h1 {
        margin: 0;
        color: white;
        padding-bottom: $padding-s;
    }
}

.login__box {
    color: $light-grey;
    background: white;
    padding: $padding-s;
    width: 80%;
    max-width: 350px;
    margin: 0 auto;
    border-radius: 20px;
    @include boxshadow;
    h2 {
        padding-bottom: $padding-s;
        font-weight: 400;
        margin: 0;
    }
    .login__box__input {
        position: relative;
        input {
            display: block;
            margin: 10px auto;
            font-size: $font-s;
            width: 90%;
            padding: 8px 8px 8px 40px;
            box-sizing: border-box;
            border-radius: 5px;
            border: $light-grey 1px solid;
            color: $off-grey;
        }
        input::placeholder {
            color: rgb(194, 194, 194);
            
        }
        i, svg {
            position: absolute;
            left: 28px;
            top: 12px;
            color: $off-purple;
        }
    }
    
    button {
        margin-top: $padding-s;
        width: 90%;
    }
}


.login__bottomBox {
    color: white;
}