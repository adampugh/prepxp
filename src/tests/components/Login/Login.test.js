import React from "react";
import { shallow } from "enzyme";

import { Login } from "../../../components/Login/Login";

describe("<Login />", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(<Login />);
        expect(wrapper).toMatchSnapshot();
    });
});
