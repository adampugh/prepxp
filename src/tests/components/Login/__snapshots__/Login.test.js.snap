// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match snapshot 1`] = `
<div>
  <div
    className="login"
  >
    <h1
      className="logo"
    >
      PrepXP
    </h1>
    <loginBox
      switchToLogin={[Function]}
      validateEmail={[Function]}
    />
  </div>
  <footer />
</div>
`;
